packages/web/src/components/gcds-error-summary/gcds-error-summary.tsx

```diff
@@ -76,15 +78,17 @@ export class GcdsErrorSummary {
   @Listen('gcdsError', { target: 'document' })
   errorListener(e) {
     if (this.listen && e.target.closest('form') == this.el.closest('form')) {
-      this.errorLinksObject[e.detail.id] = e.detail.message;
+      this.errorLinksObject[e.detail.message] = e.target;
     }
   }
 
   @Listen('gcdsValid', { target: 'document' })
   validListener(e) {
     if (this.listen && e.target.closest('form') == this.el.closest('form')) {
-      if (this.errorLinksObject && this.errorLinksObject[e.detail.id]) {
-        delete this.errorLinksObject[e.detail.id];
+      for (const [key, value] of Object.entries(this.errorLinksObject)) {
+        if (value == e.target) {
+          delete this.errorLinksObject[key];
+        }
```

Comment:
Instead of looping through the object's entries to delete keys, could you use `filter` to create a new object without the keys you want to remove? This would make it a bit more functional and easier to read.

Reply:
Since it is an object, there doesn't seem to be an easy `filter` function for it and most other ways I have found online seem to do it this way